internal/gps/source.go

```diff
@@ -635,4 +701,10 @@ type source interface {
 	disambiguateRevision(context.Context, Revision) (Revision, error)
 	exportRevisionTo(context.Context, Revision, string) error
 	sourceType() string
+	// existsRequiresList returns true if calling existsUpstream actually lists
+	// versions underneath, meaning listVersions might as well be used instead.
+	existsCallsListVersions() bool
+	// listVersionsRequiresLocal returns true if calling listVersions first
+	// requires the source to exist locally.
+	listVersionsRequiresLocal() bool
```

Comment:
These comments became out of sync.  Any better name/doc suggestions before fixing?

Reply:
these seem fine for now. there's a refactoring in store for these, anyway, i think.